.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseSetExtractor.cs

```diff
@@ -91,25 +112,105 @@ public virtual List<Token> TimeEveryday(string text, DateObject reference)
             var ret = new List<Token>();
             var ers = this.config.TimeExtractor.Extract(text, reference);
 
+            if ((config.Options & DateTimeOptions.TasksMode) != 0)
+            {
+                var ers1 = this.config.TimePeriodExtractor.Extract(text, reference);
+                if (ers.Count == 0 && ers1.Count == 1)
+                {
+                    ers = ers1;
+                }
+            }
+
             foreach (var er in ers)
             {
                 var afterStr = text.Substring(er.Start + er.Length ?? 0);
-                if ((string.IsNullOrEmpty(afterStr) || this.config.CheckBothBeforeAfter) && this.config.BeforeEachDayRegex != null)
+
+                var beforeStr = text.Substring(0, er.Start ?? 0);
+                var beforeMatch = this.config.EachDayRegex.Match(beforeStr);
+                var startIndexBeforeMatch = beforeMatch.Length + beforeMatch.Index - beforeMatch.Value.TrimStart().Length;
+                if (beforeMatch.Success)
+                {
+                    ret.Add(new Token(startIndexBeforeMatch, er.Start + er.Length ?? 0));
+                }
+
+                var match = this.config.EachDayRegex.Match(afterStr);
+                if (match.Success)
                 {
-                    var beforeStr = text.Substring(0, er.Start ?? 0);
-                    var beforeMatch = this.config.BeforeEachDayRegex.Match(beforeStr);
-                    if (beforeMatch.Success)
+                    ret.Add(new Token(er.Start ?? 0, (er.Start + er.Length ?? 0) + match.Length + match.Index));
+                }
+            }
+
+            return ret;
+        }
+
+        // Handle cases like 19th of every month
+        public virtual List<Token> DayEveryweek(string text, DateObject reference)
+        {
+            var ret = new List<Token>();
+            var ers = this.config.DateExtractor.Extract(text, reference);
+            if (NumberRecognizer.RecognizeOrdinal(text, config.Culture).Count > 0)

```

Comment:
Ideally this should use a number recognizer part of the same config.

Reply:
Also, I didn't notice this before, but I'm worried about the behaviour in languages other than English. Can we hide all of this behind the new flag and add a code comment to highlight it? Than it would be safer for now.

Reply:
I have tested the behavior of test cases in other languages and all of them are behaving the same as English. But to be on safer side for now move DayEveryweek method under flag.

Reply:
We should still need to change the call to the ordinal recognizer. But if this is the only issue, we can merge.

Reply:
At least add a `//@TODO` comment here

Reply:
added //@TODO comment.